sources/packages/web/src/views/aest/institution/Profile.vue

```diff
@@ -115,28 +104,42 @@
 
 <script lang="ts">
 import { onMounted, ref } from "vue";
+import { useRouter } from "vue-router";
+import { AESTRoutesConst } from "@/constants/routes/RouteConstants";
 import { InstitutionService } from "@/services/InstitutionService";
 import ContentGroup from "@/components/generic/ContentGroup.vue";
 import TitleValue from "@/components/generic/TitleValue.vue";
-import { AESTInstitutionDetailDto } from "@/types";
-
+import { InstitutionReadOnlyDto } from "@/types";
+import { COLOR_BLUE } from "@/constants";
+import BodyHeader from "@/components/generic/BodyHeader.vue";
 export default {
-  components: { ContentGroup, TitleValue },
+  components: { ContentGroup, TitleValue, BodyHeader },
   props: {
     institutionId: {
       type: Number,
       required: true,
     },
   },
   setup(props: any) {
-    const institutionProfileDetail = ref({} as AESTInstitutionDetailDto);
+    const institutionProfileDetail = ref({} as InstitutionReadOnlyDto);
+    const router = useRouter();
     onMounted(async () => {
-      institutionProfileDetail.value = await InstitutionService.shared.getAESTInstitutionDetailById(
+      institutionProfileDetail.value = await InstitutionService.shared.getDetail(
+        undefined,
         props.institutionId,
       );
     });
+
+    const editProfile = () => {
+      return router.push({
```

Comment:
I think you may need to await the response of router.push since navigations are asynchronous. The function needs to be async too.

Reply:
I guess we spoke about this on one of the past PR of @ann-aot. As we are not doing anything followed by push await wasn't required.

Reply:
Yes I remember, but I think that one did not return router.push. So you can just use router.push.

Reply:
Why are we returning the _router.push_?

Reply:
The return was by mistake, only in here in PR route was used with return. It has been fixed